pkg/providers/azure.go

```diff
@@ -165,17 +167,34 @@ func (sc *SetUpCmd) CreateServicePrincipal() error {
 	return nil
 }
 
-func (sc *SetUpCmd) assignSpRole() error {
+func (sc *SetUpCmd) assignSpRole(ctx context.Context) error {
 	log.Debug("Assigning contributor role to service principal...")
 
+	roleAssignClient, err := createRoleAssignmentClient(sc.SubscriptionID)
+	if err != nil {
+		return fmt.Errorf("creating role assignment client: %w", err)
+	}
+
 	scope := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", sc.SubscriptionID, sc.ResourceGroupName)
-	assignSpRoleCmd := exec.Command("az", "role", "assignment", "create", "--role", "contributor", "--subscription", sc.SubscriptionID, "--assignee-object-id", sc.spObjectId, "--assignee-principal-type", "ServicePrincipal", "--scope", scope, "--only-show-errors")
-	out, err := assignSpRoleCmd.CombinedOutput()
+	objectID := sc.spObjectId
+	roleId := "b24988ac-6180-42a0-ab88-20f7382dd24c" // Contributor role ID
```

Comment:
💡 for archival reasons here is the documented doc for this guid: https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#general